cli/internal/runcache/runcache.go

```diff
@@ -172,7 +173,9 @@ func (tc *TaskCache) RestoreOutputs(ctx context.Context, prefixedUI *cli.Prefixe
 	default:
 		// NoLogs, do not output anything
 	}
-	return cacheStatus, nil
+	// TODO: timeSaved could be part of cacheStatus, so we don't have to make a new struct
+	// downstream, but this would be a more invasive change right now.
+	return cacheStatus, timeSaved, nil
```

Comment:
I don't know how to fix this, but it's a bummer that we lose the time saved value in the most efficient case: no restoration necessary.

Maybe we put it on the daemon someday along with how long it took to generate those outputs? cc @arlyon

Reply:
Yeah, our timings could use a bit more sophistication in general. `timeSaved` doesn't account for the time it took to restore the cache at all, it's just how long the task took when it populated the cache.